### PR TITLE
fix(auth): SignOut user if the initial login state is not valid

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:*

*Description of changes:*

Keychain access was made more restrictive with PR https://github.com/aws-amplify/aws-sdk-ios/pull/4159. This would have introduced a regression where user keychain entry prior to this PR passing on to a newer device. This PR places a guard in userpool signin initialization to move to a signed out state if it could not find valid initial state.

*Check points:*

- [ ] Added new tests to cover change, if needed
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Updated CHANGELOG.md
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift [hunk 1]
```diff
@@ -274,32 +274,33 @@ final public class AWSMobileClient: _AWSMobileClient {
...
+            if userState == .signedIn
+                && (federationProvider == .userPools || federationProvider == .hostedUI)
+                && self.username == nil {
+                self.signOut()
                 currentUserState = .signedOut
-                completionHandler(.signedOut, nil)
+            } else {
+                currentUserState = userState
             }
+            completionHandler(userState, nil)
```
Comment (ameter) [inline AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift:288]: Should the completion handler be passed userState or currentUserState?

Comment (royjit) [reply AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift:288]: Thanks for pointing it out, fixed.

**Review by ameter — APPROVED** [review]
LGTM

**Review by jcjimenez — APPROVED** [review]
LGTM

**Pull request merged**